src/main/java/seedu/address/model/lab/Lab.java

```diff
@@ -57,19 +84,37 @@ public static boolean isValidLab(String test) {
      * Returns a new immutable lab with the same attributes as {@code this}.
      */
     public Lab createCopy() {
-        return new Lab(String.valueOf(labNumber), labStatus);
+        return new Lab(String.valueOf(labNumber), labStatus, labMark);
     }
 
     /**
-     * Returns a new immutable lab with the same lab number as {@code this} and the given lab status.
+     * Returns a new immutable Lab with the same lab number as {@code this} and the given lab status.
+     * The {@code LabMark} of the returned Lab is left uninitialized.
+     *
+     * @throws DuplicateLabException If the given {@code LabStatus} is equal to the existing {@code LabStatus}.
+     * @throws InvalidLabStatusException If the given {@code LabStatus} is equal to LabStatus.GRADED.
      */
-    public Lab editLabStatus(LabStatus status) throws DuplicateLabException {
+    public Lab editLabStatus(LabStatus status) throws DuplicateLabException, InvalidLabStatusException {
         if (status.equals(this.labStatus)) {
             throw new DuplicateLabException();
+        } else if (status.equals(LabStatus.GRADED)) {
+            throw new InvalidLabStatusException();
```

Comment:
Hmm I meant it such that you can't edit a `Lab`'s status to `Graded` (because you have to provide the `LabMark`), but then I realized the `Lab#of` method will throw an `InvalidLabStatusException` anyway if we try to make a `Graded` lab that way, so this part is a bit redundant. I agree that this might be quite confusing, perhaps there is a better way to implement this haha

Reply:
Alright got it!